src/main/update/util.ts

```diff
@@ -0,0 +1,47 @@
+import { DOWNLOAD_URL } from "../constants";
+import { NotSupportedPlatformError } from "../exceptions/not-supported-platform";
+import { decode, BencodexDict } from "bencodex";
+import { get as getConfig } from "../../config";
+
+export function getVersionNumberFromAPV(apv: string): number {
+  const [version] = apv.split("/");
+  return parseInt(version, 10);
+}
+
+export function decodeLocalAPV(): BencodexDict | undefined {
+  const localApvToken = getConfig("AppProtocolVersion");
+  const extra = Buffer.from(localApvToken.split("/")[1], "hex");
+
+  if (!extra.length) return;
+
+  return decode(extra) as BencodexDict | undefined;
+}
+
+export function getDownloadUrl(
+  env: string,
+  rc: number,
+  project: "player" | "launcher",
+  projectVersion: number,
+  platform: NodeJS.Platform
+): string {
+  const fn = FILENAME_MAP[platform];
+
+  if (fn === null) {
+    throw new NotSupportedPlatformError(platform);
+  }
+
+  return `${DOWNLOAD_URL}/${env}/v${rc}/${project}/v${projectVersion}/${fn}`;
+}
+
+const FILENAME_MAP: { [k in NodeJS.Platform]: string | null } = {
+  aix: null,
+  android: null,
+  darwin: "mac.tar.gz",
+  freebsd: null,
+  linux: null,
+  openbsd: null,
+  sunos: null,
+  win32: "win.zip",
+  cygwin: null,
```

Comment:
```suggestion
  cygwin: "win.zip",
```

Cygwin is a Windows environment, so this would work, but since Cygwin builds were deprecated a long time ago, I don't think this is a concern at all.

Reply:
ok 👍